glean-core/android/src/main/java/mozilla/telemetry/glean/private/MemoryDistributionMetricType.kt

```diff
@@ -0,0 +1,149 @@
+                this@MemoryDistributionMetricType.handle,
+                sample
+            )
+        }
+    }
+
+    override fun accumulateSamples(samples: LongArray) {
+        if (disabled) {
+            return
+        }
+
+        // The reason we're using [Long](s) instead of [UInt](s) in Kotlin-land is
+        // the lack of [UInt] (in stable form). The positive part of [Int] would not
+        // be enough to represent the values coming in:
+        // - [UInt.MAX_VALUE] is 4294967295
+        // - [Int.MAX_VALUE] is 2147483647
+        // - [Long.MAX_VALUE] is 9223372036854775807
+        //
+        // On the rust side, Long(s) are handled as i64 and then casted to u64.
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.launch {
+            LibGleanFFI.INSTANCE.glean_memory_distribution_accumulate_samples(
+                Glean.handle,
+                this@MemoryDistributionMetricType.handle,
+                samples,
+                samples.size
+            )
+        }
+    }
+
+    /**
+     * Tests whether a value is stored for the metric for testing purposes only. This function will
+     * attempt to await the last task (if any) writing to the the metric's storage engine before
+     * returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return true if metric value exists, otherwise false
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testHasValue(pingName: String = sendInPings.first()): Boolean {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        return LibGleanFFI
+            .INSTANCE.glean_memory_distribution_test_has_value(Glean.handle, this.handle, pingName)
+            .toBoolean()
+    }
+
+    /**
+     * Returns the stored value for testing purposes only. This function will attempt to await the
+     * last task (if any) writing to the the metric's storage engine before returning a value.
+     *
+     * @param pingName represents the name of the ping to retrieve the metric for.
+     *                 Defaults to the first value in `sendInPings`.
+     * @return value of the stored metric
+     * @throws [NullPointerException] if no value is stored
+     */
+    @VisibleForTesting(otherwise = VisibleForTesting.NONE)
+    fun testGetValue(pingName: String = sendInPings.first()): DistributionData {
+        @Suppress("EXPERIMENTAL_API_USAGE")
+        Dispatchers.API.assertInTestingMode()
+
+        if (!testHasValue(pingName)) {
+            throw NullPointerException()
+        }
+
+        val ptr = LibGleanFFI.INSTANCE.glean_memory_distribution_test_get_value_as_json_string(
+                Glean.handle,
+                this.handle,
+                pingName)!!
+
+        return DistributionData.fromJsonString(ptr.getAndConsumeRustString())!!
```

Comment:
There's a behavior change here vs. glean a-c -- the a-c version gives access to the "raw" values (i.e. without the contiguous values inserted).  I don't know if it matters, but it could cause some tests to fail (of course no one is using these distributions in third-party code yet).

Reply:
Hm, that makes it slightly more complex to test that something is empty (null or 0), but IMO us not having 2 different formats of serialization to support is better.